Compile error in profiler.cc

```
In file included from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/algorithm:62:0,
                 from /home/luotao02/Paddle/paddle/string/tinyformat/tinyformat.h:117,
                 from /home/luotao02/Paddle/paddle/string/printf.h:76,
                 from /home/luotao02/Paddle/paddle/platform/enforce.h:26,
                 from /home/luotao02/Paddle/paddle/platform/device_context.h:28,
                 from /home/luotao02/Paddle/paddle/platform/profiler.h:20,
                 from /home/luotao02/Paddle/paddle/platform/profiler.cc:15:
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/stl_algo.h: In instantiation of ‘_RandomAccessIterator std::__unguarded_partition(_RandomAccessIterator, _RandomAccessIterator, const _Tp&, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<paddle::platform::EventItem*, std::vector<paddle::platform::EventItem> >; _Tp = paddle::platform::EventItem; _Compare = std::function<bool(paddle::platform::EventItem&, paddle::platform::EventItem&)>]’:
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/stl_algo.h:2296:78:   required from ‘_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<paddle::platform::EventItem*, std::vector<paddle::platform::EventItem> >; _Compare = std::function<bool(paddle::platform::EventItem&, paddle::platform::EventItem&)>]’
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/stl_algo.h:2337:62:   required from ‘void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator = __gnu_cxx::__normal_iterator<paddle::platform::EventItem*, std::vector<paddle::platform::EventItem> >; _Size = long int; _Compare = std::function<bool(paddle::platform::EventItem&, paddle::platform::EventItem&)>]’
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/stl_algo.h:5490:44:   required from ‘void std::sort(_RAIter, _RAIter, _Compare) [with _RAIter = __gnu_cxx::__normal_iterator<paddle::platform::EventItem*, std::vector<paddle::platform::EventItem> >; _Compare = std::function<bool(paddle::platform::EventItem&, paddle::platform::EventItem&)>]’
/home/luotao02/Paddle/paddle/platform/profiler.cc:280:68:   required from here
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/stl_algo.h:2263:35: error: no match for call to ‘(std::function<bool(paddle::platform::EventItem&, paddle::platform::EventItem&)>) (paddle::platform::EventItem&, const paddle::platform::EventItem&)’
    while (__comp(*__first, __pivot))
                                   ^
In file included from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/memory:79:0,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/forward_list.h:35,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/forward_list:38,
                 from /home/luotao02/Paddle/paddle/platform/profiler.h:16,
                 from /home/luotao02/Paddle/paddle/platform/profiler.cc:15:
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/functional:2174:11: note: candidate is:
     class function<_Res(_ArgTypes...)>
           ^
In file included from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/memory:79:0,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/forward_list.h:35,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/forward_list:38,
                 from /home/luotao02/Paddle/paddle/platform/profiler.h:16,
                 from /home/luotao02/Paddle/paddle/platform/profiler.cc:15:
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/functional:2466:5: note: _Res std::function<_Res(_ArgTypes ...)>::operator()(_ArgTypes ...) const [with _Res = bool; _ArgTypes = {paddle::platform::EventItem&, paddle::platform::EventItem&}]
     function<_Res(_ArgTypes...)>::
     ^
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/functional:2466:5: note:   no known conversion for argument 2 from ‘const paddle::platform::EventItem’ to ‘paddle::platform::EventItem&’
In file included from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/algorithm:62:0,
                 from /home/luotao02/Paddle/paddle/string/tinyformat/tinyformat.h:117,
                 from /home/luotao02/Paddle/paddle/string/printf.h:76,
                 from /home/luotao02/Paddle/paddle/platform/enforce.h:26,
                 from /home/luotao02/Paddle/paddle/platform/device_context.h:28,
                 from /home/luotao02/Paddle/paddle/platform/profiler.h:20,
                 from /home/luotao02/Paddle/paddle/platform/profiler.cc:15:
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/stl_algo.h:2266:34: error: no match for call to ‘(std::function<bool(paddle::platform::EventItem&, paddle::platform::EventItem&)>) (const paddle::platform::EventItem&, paddle::platform::EventItem&)’
    while (__comp(__pivot, *__last))
                                  ^
In file included from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/memory:79:0,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/forward_list.h:35,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/forward_list:38,
                 from /home/luotao02/Paddle/paddle/platform/profiler.h:16,
                 from /home/luotao02/Paddle/paddle/platform/profiler.cc:15:
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/functional:2174:11: note: candidate is:
     class function<_Res(_ArgTypes...)>
           ^
In file included from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/memory:79:0,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/bits/forward_list.h:35,
                 from /home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/forward_list:38,
                 from /home/luotao02/Paddle/paddle/platform/profiler.h:16,
                 from /home/luotao02/Paddle/paddle/platform/profiler.cc:15:
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/functional:2466:5: note: _Res std::function<_Res(_ArgTypes ...)>::operator()(_ArgTypes ...) const [with _Res = bool; _ArgTypes = {paddle::platform::EventItem&, paddle::platform::EventItem&}]
     function<_Res(_ArgTypes...)>::
     ^
/home/luotao02/.jumbo/opt/gcc48/include/c++/4.8.3/functional:2466:5: note:   no known conversion for argument 1 from ‘const paddle::platform::EventItem’ to ‘paddle::platform::EventItem&’
make[2]: *** [paddle/platform/CMakeFiles/profiler.dir/profiler.cc.o] Error 1
make[1]: *** [paddle/platform/CMakeFiles/profiler.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/paddle/platform/profiler.cc
+++ b/paddle/platform/profiler.cc
@@ -183,35 +183,35 @@ void ParseEvents(std::vector<std::vector<Event>>& events,
   if (g_profiler_place == "") return;
 
   std::string sorted_domain;
-  std::function<bool(EventItem&, EventItem&)> sorted_func;
+  std::function<bool(const EventItem&, const EventItem&)> sorted_func;
   switch (sorted_by) {
     case EventSortingKey::kCalls:
       sorted_domain = "number of calls";
-      sorted_func = [](EventItem& a, EventItem& b) {
+      sorted_func = [](const EventItem& a, const EventItem& b) {
         return a.calls > b.calls;
       };
       break;
     case EventSortingKey::kTotal:
       sorted_domain = "total time";
-      sorted_func = [](EventItem& a, EventItem& b) {
+      sorted_func = [](const EventItem& a, const EventItem& b) {
         return a.total_time > b.total_time;
       };
       break;
     case EventSortingKey::kMin:
       sorted_domain = "minimum time";
-      sorted_func = [](EventItem& a, EventItem& b) {
+      sorted_func = [](const EventItem& a, const EventItem& b) {
         return a.min_time > b.min_time;
       };
       break;
     case EventSortingKey::kMax:
       sorted_domain = "maximum time";
-      sorted_func = [](EventItem& a, EventItem& b) {
+      sorted_func = [](const EventItem& a, const EventItem& b) {
         return a.max_time > b.max_time;
       };
       break;
     case EventSortingKey::kAve:
       sorted_domain = "average time";
-      sorted_func = [](EventItem& a, EventItem& b) {
+      sorted_func = [](const EventItem& a, const EventItem& b) {
         return a.ave_time > b.ave_time;
       };
       break;

